src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,676 @@
+                                   cbsd_grant.indoor_deployment, reliability=-1,
+                                   freq_mhz=FREQ_PROP_MODEL_MHZ)
+
+  # Compute CBSD antenna gain in the direction of protection point
+  ant_gain = antenna.GetStandardAntennaGains(incidence_angles.hor_cbsd,
+               cbsd_grant.antenna_azimuth, cbsd_grant.antenna_beamwidth,
+               cbsd_grant.antenna_gain)
+
+  # Compute FSS antenna gain in the direction of CBSD
+  fss_ant_gain = antenna.GetFssAntennaGains(incidence_angles.hor_rx, 
+                   incidence_angles.ver_rx, fss_info.pointing_azimuth, 
+                   fss_info.pointing_elevation, fss_info.max_gain_dbi)
+
+
+  # Get the total antenna gain by summing the antenna gains from CBSD to FSS
+  # and FSS to CBSD
+  effective_ant_gain = ant_gain + fss_ant_gain
+
+  # Compute EIRP of CBSD grant inside the frequency range of 
+  # protection constraint
+  eirp = getEffectiveSystemEirp(max_eirp, cbsd_grant.antenna_gain, 
+           effective_ant_gain, (cbsd_grant.high_frequency - cbsd_grant.low_frequency))
+  # Calculate the interference contribution
+  interference = eirp - getFssMaskLoss(cbsd_grant, constraint) - db_loss
+
+  return interference
+
+
+def getEffectiveSystemEirp(max_eirp, cbsd_max_ant_gain, effective_ant_gain, 
+                           reference_bandwidth=IAPBW_HZ):
+  """Calculates effective EIRP caused by a grant 
+  
+  Utility API to get effective EIRP caused by a grant in the 
+  neighborhood of the protected entity FSS/ESC/PPA/GWPZ.
+
+  Args:
+    max_eirp: The maximum EIRP allocated to the grant during IAP procedure
+    cbsd_max_ant_gain: The nominal antenna gain of the CBSD.
+    effective_ant_gain: The actual total antenna gains at the CBSD and protected entity. 
+      This takes into account the actual antenna patterns.
+    reference_bandwidth: Reference bandwidth over which effective EIRP is 
+                         calculated
+  Returns:
+    eirp_cbsd: Effective EIRP of the CBSD(dBm)
+  """
+
+  eirp_cbsd = ((max_eirp - cbsd_max_ant_gain) + effective_ant_gain + linearToDb
+            (reference_bandwidth / MHZ))
+
+  return eirp_cbsd
+
+def computeInterference(grant, eirp, channel_constraint, fss_info=None, 
+  esc_antenna_info=None, region_type=None):
+  """Calculates interference caused by a grant 
+  
+  Utility API to get interference caused by a grant in the 
+  neighborhood of the protected entity FSS/ESC/PPA/GWPZ.
+
+  Args:
+    grant : namedtuple of type CbsdGrantInformation
+    eirp : EIRP of the grant 
+    channel_constraint: namedtuple of type ProtectionConstraint
+    fss_info: contains fss point and antenna information
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain 
+    region_type: region type of protection PPA/GWPZ area.
+  Returns:
+    interference: Interference caused by a grant(dBm)
```

Comment:
description is wrong. This is not dBm as you convert it to linear.

All these routines above compute interference in dBm. Why not returning also in dBm ?
The client code can then itself do the conversion in linear before adding things up. Which he will be able to do more efficiently since it will work on a bunch of them at once

Reply:
Returning   values in dBm. Converting to linear in the caller